vortex-filtering/CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+cmake_minimum_required(VERSION 3.5)
```

Comment:
To keep things nice and tidy, you may want to split apart the CMakeLists.txt to multiple files.

That is, keep the top-level CMakeLists.txt here, but add 

- src/CMakeLists.txt (where you (potentially) add an executable)
- test/CMakeLists.txt (where you add all the test sources and link with GTest etc etc)

Then use add_subdirectory() to compose everything, and make a library in the top-level CMakeLists.txt, and do whatever with ros if that's required (although I like the idea of keeping this a library, and letting others link it into their ros-packages)

Reply:
To add to this, you don't need to have a main for testing. Instead, you can use `gtest_discover_tests(${TEST_BINARY_NAME})`, where the binary is built with add_executable, target_link_libraries as normal

Reply:
Cool. I will have to look into that